src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,145 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int capacity = INITIAL_CAPACITY;
+    private int size = 0;
+    private int threshold = (int) (capacity * LOAD_FACTOR);
+
+    MyHashMap() {
+        table = (Node<K, V>[]) new Node[capacity];
+    }
+
+    private int getHash(Object key, int capacity) {
+        return key == null ? 0 : key.hashCode() & (capacity - 1);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > threshold) {
+            resize();
+        }
+        int bucket = getHash(key,capacity);
```

Comment:
Separate getting the hash for the variable in the node and determining the position